concordium-std/src/test_infrastructure.rs

```diff
@@ -1588,6 +1648,36 @@ impl<State: Serial + DeserialWithState<TestStateApi>> TestHost<State> {
         *self.contract_balance.borrow_mut() = amount;
     }
 
+    pub fn set_self_address(&mut self, address: ContractAddress) {
+        if self.missing_contracts.contains(&address) {
+            fail!("The self_address is marked as a missing contract address.")
+        }
+        self.contract_address = Some(address);
+    }
+
+    pub fn setup_account_balance(
+        &mut self,
+        address: AccountAddress,
+        account_balance: AccountBalance,
+    ) {
+        self.query_account_balances.borrow_mut().insert(address, account_balance);
```

Comment:
Also document that the behaviour when adding the same address twice is to use the second balance.

It would not be unreasonable to expecte the balances were added. SO let's avoid confusion.